### PR TITLE
Add RARE rules for remaining theories

### PR DESCRIPTION
**Pull request by ajreynol**

### CODE REVIEW

**Comment by vleni** [comment]
Are we going to port the unit tests over as well?

**Comment by ajreynol** [comment]
Which unit tests are you referring to?

**Review by abdoo8080 — APPROVED** [review]
LGTM!

File: src/theory/uf/rewrites [hunk 1]
```diff
@@ -0,0 +1,17 @@
+; Equality
+
+(define-rule eq-refl ((t ?)) (= t t) true)
+(define-rule eq-symm ((t ?) (s ?)) (= t s) (= s t))
+
+(define-rule distinct-binary-elim ((t ?) (s ?)) (distinct t s) (not (= t s)))
+
+; bv to arith conversions
```
Comment (Lachnitt) [inline src/theory/uf/rewrites:8]: Should this really go into uf?

**Review by Lachnitt — APPROVED** [review]
Some rules are commented out but that should be fine, LGTM

**Pull request merged**